server/modules/record/service/update/thread/recordUpdateThread.js

```diff
@@ -70,7 +71,24 @@ class RecordUpdateThread extends Thread {
       this.processing = true
 
       const msg = this.queue.dequeue()
-      await this.processMessage(msg)
+      try {
+        await this.processMessage(msg)
+      } catch (e) {
+        // SystemError is an expected error type, e.g. when there's a problem with expressions.
+        if (e instanceof SystemError) {
+          this.postMessage({
+            type: WebSocketEvents.applicationError,
+            content: {
+              key: e.key,
+              params: e.params,
+            }
+          })
+          process.exit(1) // Stop processing
```

Comment:
do you really want to process.exit ?

Reply:
Right, the first time I tried to just `return`, the thread seemed to spin forever and lock the browser. But now it seems ok when I tried it again.